src/Domain/Contracts/FileProcessor.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain\Contracts;
+
+use Symfony\Component\Finder\SplFileInfo;
+
+interface FileProcessor
+{
+    public function processFile(SplFileInfo $splFileInfo): void;
```

Comment:
This is our file processor contract.
So basically if we add new static analysis tools, they get the SPLFileInfo object and can do whatever they want with that. Should make it fairly straight forward to integrate different tools